dags/stellar_etl_airflow/build_del_ins_from_gcs_to_bq_task.py

```diff
@@ -0,0 +1,204 @@
+                scope.set_extra("export_task_id", self.export_task_id)
+                scope.set_extra(
+                    "destination_table", self.destination_project_dataset_table
+                )
+                capture_message(
+                    f"failed_transforms ({self.failed_transforms}) has exceeded the max value ({self.max_failed_transforms})",
+                    "fatal",
+                )
+        super().pre_execute(context, **kwargs)
+
+
+def build_del_ins_from_gcs_to_bq_task(
+    project,
+    dataset,
+    table_id,
+    table_name,
+    export_task_id,
+    source_object_suffix,
+    partition,
+    cluster,
+    batch_id,
+    batch_date,
+    source_objects,
+    **context,
+):
+    dag = context["dag"]
+
+    # Delete operation
+
+    DELETE_ROWS_QUERY = (
+        f"DELETE FROM {dataset}.{table_name} "
+        f"WHERE batch_run_date = '{batch_date}'"
+        f"AND batch_id = '{batch_id}';"
+    )
+    delete_task = BigQueryInsertJobOperator(
+        project_id=project,
+        task_id=f"delete_old_partition_{table_name}_bq",
+        execution_timeout=timedelta(
+            seconds=Variable.get("task_timeout", deserialize_json=True)[
+                build_del_ins_from_gcs_to_bq_task.__name__
+            ]
+        ),
+        on_failure_callback=alert_after_max_retries,
+        sla=timedelta(
+            seconds=Variable.get("task_sla", deserialize_json=True)[
+                build_del_ins_from_gcs_to_bq_task.__name__
+            ]
+        ),
+        configuration={
+            "query": {
+                "query": DELETE_ROWS_QUERY,
+                "useLegacySql": False,
+            }
+        },
+    )
+    delete_task.execute(context)
+
+    # Insert operation
+
+    bucket_name = Variable.get("gcs_exported_data_bucket_name")
+    history_tables = [
+        "ledgers",
+        "assets",
+        "transactions",
+        "operations",
+        "trades",
+        "effects",
+        "contract_events",
+    ]
+
+    if cluster:
+        cluster_fields = Variable.get("cluster_fields", deserialize_json=True)
+        cluster_fields = (
+            cluster_fields[f"{table_name}"]
+            if table_id in history_tables
+            else cluster_fields[table_name]
+        )
+    else:
+        cluster_fields = None
+
+    time_partition = {}
+    if partition:
+        partition_fields = Variable.get("partition_fields", deserialize_json=True)
+        partition_fields = (
+            partition_fields[f"{table_name}"]
+            if table_id in history_tables
+            else partition_fields[table_name]
+        )
+        time_partition["type"] = partition_fields["type"]
+        time_partition["field"] = partition_fields["field"]
+
+    staging_table_suffix = ""
+    if table_name == "history_assets":
+        staging_table_suffix = "_staging"
+
+    if table_id in history_tables:
+        schema_fields = read_local_schema(f"history_{table_id}")
```

Comment:
nit: I think this can be updated to use `table_names`
Also funnily enough, setting `schema_fields` can now be done outside the if/else logic

```suggestion
        schema_fields = read_local_schema(f"{table_names}")
```

Reply:
Agreed. Moved it outside the conditional statements and defined it once outside.